### PR TITLE
[NMA-1247] Push messaging

### PR DESCRIPTION
**Pull request by Syn-McJ**
We want to be able to send push messages to Dash Wallet.

## Issue being fixed or feature implemented
Firebase Push Messaging works in 2 different ways:
- When the app is closed, the messaging works out-of-the-box and uses the notifications channel and other settings provided in the manifest.
- When the app is in the foreground, a custom service needs to be implemented and notifications are shown manually.

Additionally, FCM token has been added to the About screen for testing notifications from Firebase web interface.
There were also some unused files and config settings that I've deleted.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/Configuration.java [hunk 1]
```diff
@@ -92,9 +91,7 @@ public class Configuration {
     private static final int PREFS_DEFAULT_BTC_SHIFT = 0;
     private static final int PREFS_DEFAULT_BTC_PRECISION = 8;
     public static final String PREFS_KEY_IS_DASH_TO_FIAT_DIRECTION = "is_dash_to_fiat_direction";
-    public static final String PREFS_KEY_SEND_PAYMENT_EXCHANGE_CURRENCY = "send_payment_exchange_currency";
-    public static final String PREFS_KEY_DEFAULT_FIAT_CURRENCY_CHANGED = "fiat_currency_changed";
-    public static final String PREFS_KEY_CURRENT_FIAT_CURRENCY_CHANGED = "current_fiat_currency_changed";
```
Comment (Syn-McJ) [inline common/src/main/java/org/dash/wallet/common/Configuration.java:97]: This prefs are not used anymore.

File: liquid-integration/src/main/res/layout/activity_liquid_buy_and_sell_dash.xml [hunk 2]
```diff
@@ -113,7 +113,7 @@
                             android:layout_height="20dp"
                             android:layout_gravity="center"
                             android:layout_marginTop="4dp"
-                            app:srcCompat="@drawable/ic_dash_d_white_bottom"
+                            app:srcCompat="@drawable/ic_dash_d_white"
```
Comment (Syn-McJ) [inline liquid-integration/src/main/res/layout/activity_liquid_buy_and_sell_dash.xml:116]: There were two identical ic_dash_white. The duplicate was removed above

File: wallet/src/de/schildbach/wallet/WalletApplication.java [hunk 3]
```diff
@@ -321,6 +321,12 @@ private void createNotificationChannels() {
                 R.string.notification_synchronization_channel_name,
                 R.string.notification_synchronization_channel_description,
                 NotificationManager.IMPORTANCE_LOW);
+
+        // Push notifications
+        createNotificationChannel(getString(R.string.fcm_notification_channel_id),
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/WalletApplication.java:326]: Since the push notifications channel needs to be referenced from the manifest, we have to store it in `strings`

File: wallet/src/de/schildbach/wallet/ui/WalletActivity.java [hunk 4]
```diff
@@ -196,6 +197,15 @@ public void onChanged(de.schildbach.wallet.data.BlockchainState blockchainState)
         });
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        if (!getLockScreenDisplayed() && config.getShowNotificationsExplainer()) {
+            explainPushNotifications();
+        }
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/WalletActivity.java:206]: If it's a new install, the explainer dialog is shown when user ends up on the home screen.

File: wallet/src/de/schildbach/wallet/ui/WalletActivity.java [hunk 5]
```diff
@@ -1073,4 +1083,24 @@ private void updateSyncPaneVisibility(int id, boolean visible) {
         finish();
         return Unit.INSTANCE;
     };
+
+    @Override
+    public void onLockScreenDeactivated() {
+        if (config.getShowNotificationsExplainer()) {
+            explainPushNotifications();
+        }
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/WalletActivity.java:1091]: If it's an update, push explainer dialog shows up when the user unlocks the device.

File: common/src/main/java/org/dash/wallet/common/Configuration.java (hunk 1 above)
Comment (HashEngineering) [reply common/src/main/java/org/dash/wallet/common/Configuration.java:97]: I suppose if we resurrect bluetooth support, then we can go into the history and find the missing pieces.

File: common/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -34,6 +34,7 @@
     <string name="network_unavailable_return">Return to Wallet</string>
     <string name="button_continue">Continue</string>
     <string name="button_understand">I understand</string>
+    <string name="button_okay">Okay</string>
```
Comment (HashEngineering) [inline common/src/main/res/values/strings.xml:37]: Sometimes I wonder why "Okay" keeps showing up in our text rather than "OK", but this is unrelated to this story.  It is based on the designs.

**Review by HashEngineering — APPROVED** [review]
looks great
